sema4ai-http-helper/src/sema4ai_http/__init__.py

```diff
@@ -14,60 +16,169 @@
 
-    if enable_legacy_server_connect:
+    def _configure_tls_renegociation(self, ssl_context: ssl.SSLContext) -> None:
         if sys.version_info < (3, 12):
             _SSL_LEGACY_SERVER_CONNECT = 0x4
         else:
             _SSL_LEGACY_SERVER_CONNECT = ssl.OP_LEGACY_SERVER_CONNECT
-        ctx.options |= _SSL_LEGACY_SERVER_CONNECT
-    return ctx
+        ssl_context.options |= _SSL_LEGACY_SERVER_CONNECT
+
+
+class NetworkConfig:
+    def __init__(self) -> None:
+        self.connection_pool = self._build_connection_pool()
+
+    @staticmethod
+    def _get_network_settings_path() -> Path:
+        if sys.platform == "win32":
+            localappdata = os.environ.get("LOCALAPPDATA")
+            if not localappdata:
+                raise RuntimeError("Error. LOCALAPPDATA not defined in environment!")
+            sema4_home = Path(localappdata) / "sema4ai"
+        else:
+            # Linux/Mac
+            sema4_home = Path("~/.sema4ai").expanduser()
+
+        return sema4_home / "network-settings.yaml"
+
+    def _get_current_config_profile(self) -> dict:
+        config_file = self._get_network_settings_path()
+
+        if config_file.exists():
+            import yaml
+
+            config = yaml.safe_load(config_file.read_text())
+            current_profile = config.get("current-profile", "")
+            return next(
+                (p for p in config.get("profiles", []) if p["name"] == current_profile),
+                {},
+            )
+
+        return {}
+
+    # TODO: add support for no-proxy setting
+    def _build_connection_pool(self) -> urllib3.PoolManager | urllib3.ProxyManager:
+        profile_config = self._get_current_config_profile()
+
+        ssl_context_factory = SSLContextFactory(profile_config)
+        ssl_context = ssl_context_factory._build_ssl_context(ssl.PROTOCOL_TLS_CLIENT)
+
+        if "proxy-settings" in profile_config:
+            proxy_settings = profile_config["proxy-settings"]
+            self.connection_pool = urllib3.ProxyManager(
+                proxy_url=proxy_settings.get("https-proxy")
+                or proxy_settings.get("http-proxy"),
+                ssl_context=ssl_context,
+            )
+        else:
+            self.connection_pool = urllib3.PoolManager(ssl_context=ssl_context)
+
+        return self.connection_pool
 
 
 @lru_cache
-def _get_default_pool():
-    import ssl
+def get_network_config() -> urllib3.PoolManager | urllib3.ProxyManager:
+    return NetworkConfig().connection_pool
+
 
-    return urllib3.PoolManager(
-        ssl_context=build_ssl_context(ssl.PROTOCOL_TLS_CLIENT),
-        cert_reqs="CERT_REQUIRED",
+def build_ssl_context(
+    protocol: int | None = None, *, enable_legacy_server_connect: bool | None = None
+) -> "ssl.SSLContext":
+    return SSLContextFactory({})._build_ssl_context(
+        protocol, enable_legacy_server_connect
     )
 
 
+class ResponseWrapper:
```

Comment:
This should be a public facing class, as such, please add proper docs to it.

A TODO I had is actually document existing methods (3rd party clients have no way of knowing how the library works right now)... if you have the time, it'd be nice to fix that too (if not, that's ok, but let's try to document the API we're actually making public now).